### PR TITLE
ARROW-7219: [Python] [CI] Test with pickle5 installed

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

File: ci/docker/linux-apt-python-3.dockerfile [hunk 1]
```diff
@@ -36,7 +36,8 @@ COPY python/requirements.txt \
 RUN pip install \
     -r arrow/python/requirements.txt \
     -r arrow/python/requirements-test.txt \
-    cython setuptools
+    "pickle5; python_version == '3.6' or python_version == '3.7'" \
```
Comment (kszucs) [inline ci/docker/linux-apt-python-3.dockerfile]: Can we add this to `requirements-test.txt`?

Comment (pitrou) [reply ci/docker/linux-apt-python-3.dockerfile]: Done.

Comment (kszucs) [reply ci/docker/linux-apt-python-3.dockerfile]: Thanks!

**Review by kszucs — APPROVED** [review]
CI failure is unrelated

**Pull request closed without merging**